rogue/src/main.rs

```diff
@@ -186,11 +278,46 @@ impl State {
         vis.run_now(&self.ecs);
     }
 
-    fn run_cleanup_systems(&mut self) {
-        let mut pos = PositionMovementSystem {};
-        pos.run_now(&self.ecs);
-        DissipationSystem::clean_up_dissipated_entities(&mut self.ecs);
-        DamageSystem::clean_up_the_dead(&mut self.ecs);
+    fn run_upkeep_turn_systems(&mut self) {
+        // The ordering here is important, but for stupid reasons that should
```

Comment:
Given the amount of prose I felt necessary to write here to explain this, it's certainly the wrong solution to the problem. 🤷